### PR TITLE
DT-2096: ✨ Email user if account reactivated

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/maintain/AuthUserService.kt [hunk 1]
```diff
@@ -60,7 +61,7 @@ class AuthUserService(
     authorities: Collection<GrantedAuthority>,
   ): String {
     val email = EmailHelper.format(emailInput)
-    validate(email, firstName, lastName, EmailType.PRIMARY)
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/maintain/AuthUserService.kt:63]: this was the only usage of the method, so renamed and inline parameter

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/maintain/AuthUserService.kt [hunk 2]
```diff
@@ -227,44 +234,66 @@ class AuthUserService(
...
+    inactiveReason: String,
+    authorities: Collection<GrantedAuthority>
   ) {
     val user = userRepository.findByUsernameAndMasterIsTrue(username)
       .orElseThrow { EntityNotFoundException("User not found with username $username") }
     maintainUserCheck.ensureUserLoggedInUserRelationship(admin, authorities, user)
-    user.isEnabled = enabled
+    user.isEnabled = false
     user.inactiveReason = inactiveReason
-    // give user 7 days grace if last logged in more than x days ago
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/maintain/AuthUserService.kt:251]: not needed if disabling user

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/maintain/AuthUserService.kt [hunk 3]
```diff
@@ -227,44 +234,66 @@ class AuthUserService(
 
   @Transactional(transactionManager = "authTransactionManager")
   @Throws(AuthUserGroupRelationshipException::class)
-  fun enableUser(usernameInDb: String, admin: String, authorities: Collection<GrantedAuthority>) =
-    changeUserEnabled(usernameInDb, true, null, admin, authorities)
+  fun enableUser(username: String, admin: String, requestUrl: String, authorities: Collection<GrantedAuthority>) {
+    val user = userRepository.findByUsernameAndMasterIsTrue(username)
+      .orElseThrow { EntityNotFoundException("User not found with username $username") }
+    maintainUserCheck.ensureUserLoggedInUserRelationship(admin, authorities, user)
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/maintain/AuthUserService.kt:240]: separated out code into two separate calls so can add email for activation scenario

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/maintain/AuthUserService.kt [hunk 4]
```diff
@@ -355,10 +384,8 @@ class AuthUserService(
     private val log: Logger = LoggerFactory.getLogger(this::class.java)
 
     // Data item field size validation checks
-    private const val MAX_LENGTH_USERNAME = 30
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/maintain/AuthUserService.kt:358]: unused

File: src/main/resources/application.yml [hunk 5]
```diff
@@ -83,15 +83,16 @@ application:
 
   notify:
     verify.template: 4b285841-41ba-47d1-b6b0-104440c4e312
+    verify-mobile.template: b50619eb-6e1f-4348-b231-d38c01c82c85
```
Comment (petergphillips) [inline src/main/resources/application.yml:86]: moved round so that like templates are next to one another in config

**Review by SimonMitchellMOJ — APPROVED** [review]
LGTM

**Review by SimonMitchellMOJ — APPROVED** [review]
LGTM

**Pull request merged**